### PR TITLE
Ignore empty references when added to ReferenceList

### PR DESCRIPTION
**Pull request by JeroenDeDauw**

### CODE REVIEW

File: src/ReferenceList.php [hunk 1]
```diff
@@ -70,6 +70,17 @@ public function addReference( Reference $reference, $index = null ) {
 		}
 	}
 
+	/**
+	 * @see parent::attach
```
Comment (thiemowmde) [inline src/ReferenceList.php]: `@see SplObjectStorage::attach` please.

File: src/ReferenceList.php [hunk 2]
```diff
@@ -70,6 +70,17 @@ public function addReference( Reference $reference, $index = null ) {
...
+	 * @param Reference $reference
+	 * @param mixed $data
+	 */
+	public function attach( $reference, $data = null ) {
+		if ( !$reference->isEmpty() ) {
```
Comment (thiemowmde) [inline src/ReferenceList.php]: I think you need to add this check to `addReference`. Calling `addReference( new Reference(), 0 )` will still be allowed but shouldn't for consistency.

File: tests/unit/ReferenceListTest.php [hunk 3]
```diff
@@ -141,41 +142,61 @@ public function testRemoveReference( ReferenceList $array ) {
...
+
+	public function testGivenEmptyReference_addReferenceDoesNotAdd() {
+		$reference1 = new Reference( array( new PropertyNoValueSnak( 1 ) ) );
+		$reference2 = new Reference( array( new PropertyNoValueSnak( 2 ) ) );
+		$emptyReference = new Reference( array() );
 
-		$array->rewind();
-		$array->next();
+		$references = new ReferenceList( array( $reference1, $reference2 ) );
+		$references->addReference( $emptyReference );
```
Comment (thiemowmde) [inline tests/unit/ReferenceListTest.php]: As hinted above, please add the same test with `addReference( ..., 0 )`.

File: src/ReferenceList.php (hunk 1 above)
Comment (JeroenDeDauw) [inline src/ReferenceList.php]: Done

File: tests/unit/ReferenceListTest.php (hunk 3 above)
Comment (JeroenDeDauw) [inline tests/unit/ReferenceListTest.php]: I've added the test to verify the behaviour is as expected. The bug you hinted at appears to not be there

**Comment by JeroenDeDauw** [comment]
New version in #399. For some reason having problems pushing to this one.

> git push origin empyrefs -f
error: src refspec empyrefs does not match any.
error: failed to push some refs to 'git@github.com:wmde/WikibaseDataModel.git'

File: tests/unit/ReferenceListTest.php (hunk 3 above)
Comment (thiemowmde) [inline tests/unit/ReferenceListTest.php]: Oh, right. I double-checked and addReference depends on append in all cases, with and without index.

**Pull request closed without merging**